cmake/compilers/Clang.cmake

```diff
@@ -12,6 +12,14 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+if(CMAKE_CXX_COMPILER MATCHES "/em\\+\\+(-[a-zA-Z0-9.])?$")
+  message(" * C++ compiler: Emscripten")
+  set(CMAKE_EXECUTABLE_SUFFIX .js)
+  set(TBB_EMSCRIPTEN 1)
+  set(CMAKE_CXX_FLAGS ${CMAKE_CXX_FLAGS} " -pthread -s TOTAL_MEMORY=128MB -s EXIT_RUNTIME=1 -s PROXY_TO_PTHREAD=1")
```

Comment:
I don't recommend setting the link-time flags `-s TOTAL_MEMORY=128MB -s EXIT_RUNTIME=1 -s PROXY_TO_PTHREAD=1` here. Client apps that consume oneTBB may prefer different settings. Only `-pthread` is truly needed here. The other flags should probably be set where the executable targets are defined. For example, the `tbb_add_test()` function in `test/CMakeLists.txt` could apply these settings to each executable test using CMake's `target_link_options()` function.

Also, instead of guessing how much total memory is required, you may wish to use `-s ALLOW_MEMORY_GROWTH=1`. In that case, you can suppress the compiler warning about using threads and growable memory with the `-Wno-pthreads-mem-growth` flag.